### PR TITLE
Convert `SuperSelect` components to TypeScript

### PR DESCRIPTION
**Pull request by alex-ju**
### :pushpin: Summary

Convert `SuperSelect` components to TypeScript

### :hammer_and_wrench: Detailed description

- [x] Converted `SuperSelect::Single` components to TypeScript
- [x] Converted `SuperSelect::Multiple` components to TypeScript

Notable changes:
  - moved from extending the `PowerSelectComponent` class to using the `publicAPI` (this is because we now need to extend the glimmer `Component`; it also makes more sense to uncouple the classes and use the API instead)
  - fixed default results message in `SuperSelect::Single` (changed from `results` to `total` as per design)

### :link: External links

Jira ticket: [HDS-2723](https://hashicorp.atlassian.net/browse/HDS-2723)

***

### 👀 Component checklist

- [x] Percy was checked for any visual regression
- [x] A changelog entry was added via [Changesets](https://github.com/changesets/changesets) if needed (see [templates here](https://github.com/hashicorp/design-system/blob/main/wiki/Website-Changelog.md#templates-for-npm-packages))

:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

[HDS-2723]: https://hashicorp.atlassian.net/browse/HDS-2723?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .changeset/afraid-guests-relax.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@hashicorp/design-system-components": minor
+---
+
+Converted `SuperSelect` components to TypeScript
```
Comment (shleewhite) [inline .changeset/afraid-guests-relax.md:5]: Nit: most of the other ts conversion change sets have been like:
```md
`Table` - Converted component and sub-components to TypeScript
```

File: packages/components/src/components/hds/form/super-select/types.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+export type HdsFormSuperSelectHorizontalPositions =
+  `${HdsFormSuperSelectHorizontalPositionValues}`;
+
+// map SuperSelect's `horizontalPosition` values to anchoredPositionModifier's `placement` values
+export const HdsFormSuperSelectHorizontalPositionToPlacementValues: Record<
+  // SuperSelect's `horizontalPosition` values
+  HdsFormSuperSelectHorizontalPositionValues,
+  FloatingUIOptions['placement']
+> = {
+  [HdsFormSuperSelectHorizontalPositionValues.Left]: 'bottom-start',
```
Comment (shleewhite) [inline packages/components/src/components/hds/form/super-select/types.ts:23]: Praise: very slick

File: packages/components/src/components/hds/form/super-select/single/base.ts [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+    return this.args.searchPlaceholder ?? 'Search';
+  }
+
+  /**
+   * Get the maxWidth to apply to the dropdown
+   * @param dropdownMaxWidth
+   * @type {string}
+   * @default 'none'
+   */
+  get dropdownMaxWidthStyle() {
```
Comment (shleewhite) [inline packages/components/src/components/hds/form/super-select/single/base.ts:115]: Nit: missing this return type

File: packages/components/src/components/hds/form/super-select/multiple/base.ts [hunk 4]
```diff
@@ -109,7 +141,7 @@ export default class HdsSuperSelectMultipleBaseComponent extends PowerSelectComp
    * @default 'none'
    */
   get dropdownMaxWidthStyle() {
```
Comment (shleewhite) [inline packages/components/src/components/hds/form/super-select/multiple/base.ts:143]: Nit: missing this return type

File: packages/components/src/components/hds/form/super-select/multiple/base.ts [hunk 5]
```diff
@@ -3,46 +3,78 @@
...
-      placement: this.args.horizontalPosition
-        ? HORIZONTAL_POSITION_MAPPING[this.args.horizontalPosition]
-        : DEFAULT_HORIZONTAL_POSITION,
+      placement: HORIZONTAL_POSITION_MAPPING[this.horizontalPosition],
       offsetOptions: 4,
       enableCollisionDetection: true,
     });
     // prevent PowerSelect from setting position
-    return {};
+    return {
```
Comment (shleewhite) [inline packages/components/src/components/hds/form/super-select/multiple/base.ts:73]: Out of curiosity, why did the return value change?

File: packages/components/src/components/hds/form/super-select/multiple/field.hbs [hunk 6]
```diff
@@ -65,7 +63,7 @@
       @triggerComponent={{@triggerComponent}}
       @triggerId={{F.id}}
       @triggerRole={{@triggerRole}}
-      @typeAheadMatcher={{@typeAheadMatcher}}
+      @typeAheadOptionMatcher={{@typeAheadOptionMatcher}}
```
Comment (shleewhite) [inline packages/components/src/components/hds/form/super-select/multiple/field.hbs:66]: Out of curiosity, why did this get renamed?

File: packages/components/src/components/hds/form/super-select/single/field.hbs [hunk 7]
```diff
@@ -12,8 +11,7 @@
   as |F|
 >
   {{! Notice: the order of the elements is not relevant here, because is controlled at "Hds::Form::Field" component level }}
-  {{yield (hash Label=F.Label isRequired=F.isRequired isOptional=F.isOptional)}}
-  {{yield (hash HelperText=F.HelperText Error=F.Error)}}
+  {{yield (hash Label=F.Label HelperText=F.HelperText Error=F.Error)}}
```
Comment (shleewhite) [inline packages/components/src/components/hds/form/super-select/single/field.hbs:14]: Do you not need to pass isRequired/isOptional anymore?

**Review by shleewhite — COMMENTED** [review]
Mostly questions! Couple of small things and I think you need to add an export to the `components.ts` file.

File: packages/components/src/components/hds/form/super-select/multiple/field.hbs (hunk 6 above)
Comment (alex-ju) [reply packages/components/src/components/hds/form/super-select/multiple/field.hbs:66]: The argument is [wrongly documented as `@typeAheadMatcher`](https://ember-power-select.com/docs/api-reference). [In practice, it is `@typeAheadOptionMatcher`](https://github.com/cibernox/ember-power-select/blob/master/ember-power-select/src/components/power-select.ts#L117). I'll raise this with the author to fix the docs.

File: packages/components/src/components/hds/form/super-select/single/field.hbs (hunk 7 above)
Comment (alex-ju) [reply packages/components/src/components/hds/form/super-select/single/field.hbs:14]: [We recently discovered that these were not needed](https://github.com/hashicorp/design-system/pull/2240/files#diff-9b906f806254a94975bab30d6ba8378b8d41572aff5eac578645ec7911bfff04R7), as they are part of the label/legend API

File: packages/components/src/components/hds/form/super-select/multiple/base.ts (hunk 5 above)
Comment (alex-ju) [reply packages/components/src/components/hds/form/super-select/multiple/base.ts:73]: We previously returned an empty object, forcing the PowerSelect to fall back to the default values. Now that we moved to TypeScript we need to provide the bare minimum for the `CalculatePositionResult` object to have a type match. I provided some default values for the mandatory properties, but the computation is done by the `anchoredPositionModifier` so it doesn't make much of a difference.

File: packages/components/src/components/hds/form/super-select/single/base.ts (hunk 3 above)
Comment (alex-ju) [reply packages/components/src/components/hds/form/super-select/single/base.ts]: Good catch! Added in https://github.com/hashicorp/design-system/pull/2356/commits/08c99afae38f74beb19294c618baf00e1eb2ee5e

File: packages/components/src/components/hds/form/super-select/multiple/base.ts (hunk 4 above)
Comment (alex-ju) [reply packages/components/src/components/hds/form/super-select/multiple/base.ts]: Good catch! Added in https://github.com/hashicorp/design-system/pull/2356/commits/08c99afae38f74beb19294c618baf00e1eb2ee5e

File: .changeset/afraid-guests-relax.md (hunk 1 above)
Comment (alex-ju) [reply .changeset/afraid-guests-relax.md]: Updated in https://github.com/hashicorp/design-system/pull/2356/commits/217c865c29b101cebdb60b7f19fd91be31f57dac

**Review by shleewhite — APPROVED** [review]

**Pull request merged**